### PR TITLE
adds serving in admin to show services, revisions, routes as tab views

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4880

**Analysis / Root cause**: 
Admin should see Serving under Serverless and all resources (services, revisions, routes) under it

**Solution Description**: 
- moved Serving under Serverless and all resources (services, revisions, routes) under it

**Screen shots / Gifs for design review**: 

image

Gif:

Oct-07-2020 19-45-26

@openshift/team-devconsole-ux

**Unit test coverage report**: 

image

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/knative-plugin/src/plugin.tsx [hunk 1]
```diff
@@ -383,6 +363,30 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+  },
+  {
+    type: 'Page/Route',
+    properties: {
+      exact: false,
+      path: ['/serving/all-namespaces', '/serving/ns/:ns'],
+      loader: async () =>
+        (
+          await import(
+            './components/overview/serving-list/ServingListsPage' /* webpackChunkName: "admin-pipeline" */
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/plugin.tsx:385]:
```suggestion
            './components/overview/serving-list/ServingListsPage' /* webpackChunkName: "serving-list-page" */
```

File: frontend/packages/knative-plugin/src/components/overview/serving-list/ServingListsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { Page } from '@console/internal/components/utils';
+import { NamespaceBar } from '@console/internal/components/namespace';
+import { MenuActions } from '@console/shared/src/components/multi-tab-list/multi-tab-list-page-types';
+import MultiTabListPage from '@console/shared/src/components/multi-tab-list/MultiTabListPage';
+import ServicesPage from '../../services/ServicesPage';
+import RevisionsPage from '../../revisions/RevisionsPage';
+import RoutesPage from '../../routes/RoutesPage';
+
+interface ServingListPageProps {
+  match: Rmatch<any>;
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/components/overview/serving-list/ServingListsPage.tsx:14]: nit:
```suggestion
  match: Rmatch<{ ns: string }>;
```

File: frontend/packages/dev-console/src/components/pipelines-lists/PipelinesListsPage.tsx [hunk 3]
```diff
@@ -10,11 +9,16 @@ import {
 } from '../../models';
 import { Page } from '@console/internal/components/utils';
 import { TechPreviewBadge } from '@console/shared';
+import {
+  MenuAction,
+  MenuActions,
+} from '@console/shared/src/components/multi-tab-list/multi-tab-list-page-types';
+import MultiTabListPage from '@console/shared/src/components/multi-tab-list/MultiTabListPage';
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/pipelines-lists/PipelinesListsPage.tsx:16]: should import from `@console/shared`
```suggestion
import {
  MenuAction,
  MenuActions,
  MultiTabListPage
} from '@console/shared';
```

**Review by sahil143 — CHANGES_REQUESTED** [review]

**Comment by bmignano** [comment]
LGTM!

**Review by sahil143 — APPROVED** [review]
/lgtm

**Review by vikram-raj — COMMENTED** [review]
I notice one issue, clicking on the details page breadcrumb should redirect back to the same tabbed list page. But currently, it is redirected back to the default list page.

**Comment by invincibleJai** [comment]
> I notice one issue, clicking on the details page breadcrumb should redirect back to the same tabbed list page. But currently, it is redirected back to the default list page.

@vikram-raj thanks for highlighting, have updated PTAL

**Review by vikram-raj — COMMENTED** [review]
Thanks @invincibleJai.

/lgtm

**Review by rohitkrai03 — COMMENTED** [review]
Code looks good to me but found this bug that may not be related to this PR. Please take a look - 

Peek 2020-10-15 17-15

**Comment by invincibleJai** [comment]
> Code looks good to me but found this bug that may not be related to this PR. Please take a look -
> 

Thanks @rohitkrai03 Similar behaviour is there for other flows but yes there was an issue should show `create Service`. Have fixed it PTAL

Oct-19-2020 19-52-25

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Review by sahil143 — APPROVED** [review]
/lgtm

**Pull request merged**